File: baekjoon/2897/main.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
    int r, c;
    cin >> r >> c;

    char map[50][50];
    string row;
    for (int i = 0; i < r; i++) {
        cin >> row;
        for (int j = 0; j < row.size(); j++) {
            map[i][j] = row[j];
        }
    }

    int answer[5] = { 0, 0, 0, 0, 0 };
    for (int i = 0; i < r - 1; i++) {
        for (int j = 0; j < c - 1; j++) {
            string space = "";
            space.push_back(map[i][j]);
            space.push_back(map[i][j+1]);
            space.push_back(map[i+1][j]);
            space.push_back(map[i+1][j+1]);
            
            if (space.find('#') == string::npos)
                answer[count(space.begin(), space.end(), 'X')]++;
        }
    }

    for (int i = 0; i < 5; i++)
        cout << answer[i] << '\n';

    return 0;
}

File: baekjoon/2897/go/main.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var r, c int
	fmt.Scanf("%d %d\n", &r, &c)

	m := make([][]rune, r)
	for i := range m {
		m[i] = make([]rune, c)
	}

	for i := 0; i < r; i++ {
		var row string
		fmt.Scanf("%s\n", &row)
		for j := range row {
			m[i][j] = rune(row[j])
		}
	}

	answer := [5]int{0, 0, 0, 0, 0}
	for i := 0; i < r-1; i++ {
		for j := 0; j < c-1; j++ {
			var space string
			space += string(m[i][j]) + string(m[i][j+1]) + string(m[i+1][j]) + string(m[i+1][j+1])

			if strings.Count(space, "#") < 1 {
				answer[strings.Count(space, "X")]++
			}

		}
	}

	for i := 0; i < 5; i++ {
		fmt.Println(answer[i])
	}
}
